cmd/csi-api-gen/generators/api.go

```diff
@@ -0,0 +1,432 @@
+package generators
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"sort"
+	"strings"
+
+	"k8s.io/gengo/args"
+	conversiongenerator "k8s.io/gengo/examples/conversion-gen/generators/generator"
+	"k8s.io/gengo/generator"
+	"k8s.io/gengo/namer"
+	"k8s.io/gengo/types"
+	"k8s.io/klog"
+
+	"github.com/kubernetes-csi/csi-proxy/client/apiversion"
+)
+
+/*
+This file defines a gengo generator to generate all the boilerplate code API
+groups and versions need.
+
+See this repo's client/api/README.md for more context.
+*/
+
+const (
+	// csiProxyRootPath is the CSI-proxy go library's root path.
+	csiProxyRootPath = "github.com/kubernetes-csi/csi-proxy"
```

Comment:
Sure, though I'm not sure what the use case would be?

Reply:
I was thinking it will make name changes easier to handle in the future. Looks like k/k hardcodes this as well in their generators: https://github.com/kubernetes/kubernetes/blob/a3ccea9d8743f2ff82e41b6c2af6dc2c41dc7b10/staging/src/k8s.io/code-generator/cmd/conversion-gen/generators/conversion.go#L439-L440. So fine for now.

Reply:
But even then, that parameter presumably would have to be hardcoded somewhere in this repo? :)

Keeping as is for now anyway, as agreed.